plugins/BEdita/Core/src/Model/Table/ObjectTypesTable.php

```diff
@@ -186,13 +195,29 @@ public function get($primaryKey, $options = [])
     }
 
     /**
+     * Set default parent on creation if missing.
+     *
+     * @param \Cake\Event\Event $event The event dispatched
+     * @param \Cake\Datasource\EntityInterface $entity The entity to save
+     * @return void
+     */
+    public function beforeSave(Event $event, EntityInterface $entity)
+    {
+        if ($entity->isNew() && empty($entity->parent_id)) {
+            $entity->parent_id = self::DEFAULT_PARENT_ID;
+        }
+    }
+
+    /**
      * Invalidate cache after saving an object type.
+     * Recover Nested Set Model tree structure (tree_left, tree_right)
      *
      * @return void
      */
     public function afterSave()
     {
         Cache::clear(false, self::CACHE_CONFIG);
+        $this->behaviors()->get('Tree')->recover();
```

Comment:
Is this required? Since Tree behavior is attached, any operation that invokes callbacks (i.e.: `save()`, `saveMany()`, `delete()`, …) should already result in a valid tree. However, `updateAll()` and `deleteAll()` still require this… but those do not fire the `afterSave` event either. 🤔

Reply:
Honestly I don't know, but if I comment this line `tree_left` and `tree_right` are not correctly updated.
I'm sure there's a better way to do it.... but at least it works